src/botPage/bot/Interpreter.js

```diff
@@ -90,11 +92,16 @@ export default class Interpreter {
                 globalObserver.emit('Error', e);
                 const { initArgs, tradeOptions } = this.bot.tradeEngine;
                 this.stop();
-                this.init();
-                this.$scope.observer.register('Error', onError);
-                this.bot.tradeEngine.init(...initArgs);
-                this.bot.tradeEngine.start(tradeOptions);
-                this.revert(this.startState);
+                const unrecoverableErrorList = ['InsufficientBalance'];
```

Comment:
You can move this to https://github.com/gokula-krishna-dev/binary-bot/blob/b62ca069468e692f6f72bc6a023ebb276f240789/src/botPage/bot/Interpreter.js#L88, and add the check for `CustomLimitsReached`.

Reply:
I would prefer an optional error argument for `shouldRestartOnError` instead:

```
const unrecoverableErrorList = ['InsufficientBalance', 'CustomLimitsReached'];

const shouldRestartOnError = (bot, e = {}) => !unrecoverableErrorList.includes(e.name)  && botInitialized(bot) && bot.tradeEngine.options.shouldRestartOnError;

...

if (!shouldRestartOnError(this.bot, e) || !botStarted(this.bot)) {
  reject(e);
  return;
}

```